service/template.go

```diff
@@ -65,13 +65,19 @@ func getTemplate(w http.ResponseWriter, r *http.Request, envId string) (int, err
 
 	catalogName, templateName, templateBase, revisionOrVersion, _ := parse.TemplateURLPath(catalogTemplateVersion)
 
+	environment := r.URL.Query().Get("projectId")
```

Comment:
Looking at this more close, I think the proper place for this logic is here:
https://github.com/rancher/catalog-service/blob/master/service/utils.go#L21
that way, you have one centralized place where env Id is retrieved. The new logic in there should be that if it is not found as a header, look for it as the projectId query parameter. If its not found in either, then return an error as it is doing today.